BrightID/src/components/Connections/ConnectionScreenController.js

```diff
@@ -25,35 +25,38 @@ function ConnectionScreenController(props: ConnectionScreenProps) {
   );
   const myConnections = useSelector((state) => state.connections.connections);
   const myGroups = useSelector((state) => state.groups.groups);
+  const [connectionsNum, setConnectionsNum] = useState(0);
+  const [groupsNum, setGroupsNum] = useState(0);
   const [mutualGroups, setMutualGroups] = useState<Array<group>>([]);
-  const [mutualConnections, setMutualConnections] = useState<Array<connection>>(
-    [],
-  );
-  const [verifications, setVerifications] = useState<Array<string>>([]);
+  const [mutualConnections, setMutualConnections] = useState<Array<connection>>([]);
+  const [verified, setVerified] = useState(false);
+  const [createdAt, setCreatedAt] = useState(0);
+  const [connectedAt, setConnectedAt] = useState(0);
   const [loading, setLoading] = useState(false);
 
   useFocusEffect(
     useCallback(() => {
       const fetchData = async (connectionId) => {
         setLoading(true);
         console.log(`fetching connection info for ${connectionId}`);
-        const connectionData = await fetchConnectionInfo({
-          brightId: connectionId,
-          myConnections,
-          myGroups,
+        const profile = await api.getUserProfile(connectionId);
+        console.log(profile, 22);
+        setConnectionsNum(profile.connectionsNum);
+        setGroupsNum(profile.groupsNum);
+        setVerified(profile.verified);
+        setCreatedAt(profile.createdAt);
+        setConnectedAt(profile.connectedAt);
+        profile.mutualConnections = myConnections.filter(function (conn) {
```

Comment:
I found it super confusing that `profile` suddenly includes an array of objects, although the API just returns an array of IDs. Especially because the debugger console does lazy evaluation of the logged object, so when you expand the response in the the console log it shows the mapped objects instead of the original response from the API.

Maybe just skip the local variable alltogether and just do
```
setMutualConnections(myConnections.filter(function (conn) {
    return profile.mutualConnections.includes(conn.id);
￼});